src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -125,7 +125,8 @@ public async Task<ReturnResult> DeleteItemsAsync(IEnumerable<IStorageItemWithPat
 			var deleteFromRecycleBin = source.Select(item => item.Path).Any(path => RecycleBinHelpers.IsPathUnderRecycleBin(path));
 			var canBeSentToBin = !deleteFromRecycleBin && await RecycleBinHelpers.HasRecycleBin(source.FirstOrDefault()?.Path);
 
-			if (showDialog)
+			if (showDialog is DeleteConfirmationPolicies.Always
+				|| showDialog is DeleteConfirmationPolicies.PermanentOnly && (permanently || !canBeSentToBin))
```

Comment:
```suggestion
				|| showDialog is DeleteConfirmationPolicies.PermanentOnly &&
				(permanently || !canBeSentToBin))
```

Reply:
I don't think it is better to separate this line because `(permanently || !canBeSentToBin))` is integral to `showDialog is DeleteConfirmationPolicies.PermanentOnly`

Reply:
Okay, feel free to revert the change

Reply:
I will revert. Thank you.